agents/timestream_aggregator/timestream_aggregator.py

```diff
@@ -243,6 +255,24 @@ def read_frames(self):
 
         self.frames = self.reader.Process(None)
         if self.frames:
+            # Handle flow control frames
+            for f in self.frames:
+                if f.type in [core.G3FrameType.none]:
+                    if 'sostream_flowcontrol' not in f:
+                        self.log.warn("Improperly formatted flow control " +
+                                      "frame encountered.")
+
+                    flow = f.get('sostream_flowcontrol')
+
+                    # START; create_new_file
+                    if FlowControl(flow) is FlowControl.START:
```

Comment:
Won't this break here if f.get('sostream_flowcontrol') returned None? Probably don't have to use a get call, since it would break anyway if the key doesn't exist.

Reply:
Ah, yeah, good catch. I added a continue after the check and warning about improperly formatted flow control frames.